File: week01/YOOKYUNG/1206.py
for tc in range(1, 11):
    N = int(input())
    arr = list(map(int, input().split()))

    result = 0

# 처음부터 하지 않는 이유 : 처음은 필요 없음 N-2
    for i in range(2, N - 2):
        diff = arr[i] - max(arr[i-2], arr[i-1], arr[i+1], arr[i+2])

        if diff > 0:
            result += diff

    print(f"#{tc} {result}")

File: week01/YOOKYUNG/1206.cpp
#include <iostream>
#include <algorithm>
using namespace std;

int main()
{
  for (int tc = 1; tc <= 10; tc++)
  {
    int N;
    cin >> N;

    int arr[1000];

    // 건물 높이 입력받기
    for (int i = 0; i < N; i++)
    {
      cin >> arr[i];
    }

    int result = 0;

    // 양쪽 2칸씩 비교해야 해서 2부터 N-3까지 확인
    for (int i = 2; i < N - 2; i++)
    {

      // 주변 건물 중 가장 높은 건물 찾기
      int maxheight = max(max(arr[i - 2], arr[i - 1]),
                          max(arr[i + 1], arr[i + 2]));
      int diff = arr[i] - maxheight;
      if (diff > 0)
      {
        result += diff;
      }
    }

    cout << "#" << tc << " " << result << endl;
  }

  return 0;
}